maintain state of recording starting in db for daily

zach-at-the-skills opened the pull request.

zach-at-the-skills commented:
Only issue I really found is that sometimes the cloud recording fails unexpectedly, but our code is unaware of that and marks it as started regardless. There isn't really a great solution around that, though.

mcasey8540 commented on pages/call/[uid].tsx:
```diff
@@ -65,10 +65,27 @@ export default function JoinCall(props, session) {
+        })
+        .then(() => {
+          if (!props.booking.dailyRef.recordingHasStarted) {
+            callFrame.startRecording({
```
@zach-at-the-skills - does `callFrame.startRecording` return a promise? if so, would be nice to put the fetch inside the `then` block.

zach-at-the-skills replied:
![image](https://user-images.githubusercontent.com/75326654/148972237-7bcb4456-92e4-4de6-93c2-c02ed9629cae.png)
unfortunately it doesn't return a promise. No return (void).

mcasey8540 commented on pages/call/[uid].tsx:
```diff
@@ -100,15 +117,20 @@ export default function JoinCall(props, session) {
-              max_cam_streams: 5,
-            },
-          });
+          if (!props.booking.dailyRef.recordingHasStarted) {
```
Opportunity to DRY up these similar code blocks

mcasey8540 committed:
CU-2447awn - use query param record to start recording

Notes: The recording will not stop if the user with the query param record leaves the call. If the user with the query param record rejoins the call, they will see an error message, but the recording is technically still happening
